library/std/src/os/linux/mod.rs

```diff
@@ -4,5 +4,9 @@
 #![doc(cfg(target_os = "linux"))]
 
 pub mod fs;
+mod net;
 pub mod process;
 pub mod raw;
+
+#[unstable(feature = "tcp_quickack", issue = "96256")]
+pub use net::TcpStreamExt;
```

Comment:
You're still exporting `TcpStreamExt` directly from `os::linux` rather than exporting the `net` module which then contains `TcpStreamExt`.

You might want to build the documentation locally with `x.py doc library/std --open` and check that everything is in the right place.